### ISSUE TITLE
Failing getting CUDA code

### ISSUE THREAD

**Issue by davideschiavone** [brian-team/brian2cuda#260]
Dear all, 

I modified my code setting the device as suggested in the README to use CUDA. 

Here is my code to get familiar with CUDA:

https://github.com/davideschiavone/SimpleMNIST/blob/main/mnist_example.py

However, I get error in the generated cpp file. Attached you find the log [log.txt](https://github.com/brian-team/brian2cuda/files/7896071/log.txt)

My nvcc version is:

````
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Wed_Jun__2_19:15:15_PDT_2021
Cuda compilation tools, release 11.4, V11.4.48
Build cuda_11.4.r11.4/compiler.30033411_0
````

**Comment by denisalevi** [comment]
Hi @davideschiavone,

there is currently an issue with CUDA 11.4 which I haven't fixed yet, see #224. Might be a quick fix actually, I just haven't done it yet since I'm running benchmarks for a paper on CUDA 11.2 currently. I'll have a look later if this is fixed easily, else try CUDA 11.2 for now.

Let me know if it works :)

**Comment by davideschiavone** [comment]
Thanks @denisalevi for the quick reply.

That solved my issue
